storybook/src/components/Icon/Icon.stories.tsx

```diff
@@ -83,60 +69,31 @@ export const WithLargeBodyText: Story = {
   ),
 }
 
-export const WithHeading3: Story = {
+export const WithAHeading: Story = {
   args: {
     size: 'heading-3',
   },
-  render: (args) => (
-    <Row gap="small">
-      <Icon {...args} />
-      <Heading color={args.color} level={3}>
-        Heading 3 text
-      </Heading>
-    </Row>
-  ),
-}
-
-export const WithHeading4: Story = {
-  args: {
-    size: 'heading-4',
-  },
-  render: (args) => (
-    <Row gap="small">
-      <Icon {...args} />
-      <Heading color={args.color} level={4}>
-        Heading 4 text
-      </Heading>
-    </Row>
-  ),
-}
-
-export const WithHeading5: Story = {
-  args: {
-    size: 'heading-5',
+  argTypes: {
+    size: {
+      options: [...iconSizes.filter((size) => size.startsWith('heading-'))],
+    },
   },
-  render: (args) => (
-    <Row gap="small">
-      <Icon {...args} />
-      <Heading color={args.color} level={4} size="level-5">
-        Heading 5 text
-      </Heading>
-    </Row>
-  ),
-}
+  render: (args) => {
+    const headingSize = args.size ? (args.size.slice(-1) as '0' | '1' | '2' | '3' | '4' | '5' | '6') : '3'
```

Comment:
Good idea! Should we do something similar for the body text sizes? You can change the size there now, without the accompanying text size changing.